backend/tracim_backend/models/contents.py

```diff
@@ -71,50 +71,44 @@ def __init__(
 )
 
 
-CONTENT_DEFAULT_STATUS = [
-    open_status,
-    closed_validated_status,
-    closed_unvalidated_status,
-    closed_deprecated_status,
-]
+class ContentStatusList(object):
 
+    OPEN = open_status
 
-class ContentStatusLegacy(NewContentStatus):
-    """
-    Temporary remplacement object for Legacy ContentStatus Object
-    """
-    OPEN = open_status.slug
-    CLOSED_VALIDATED = closed_validated_status.slug
-    CLOSED_UNVALIDATED = closed_unvalidated_status.slug
-    CLOSED_DEPRECATED = closed_deprecated_status.slug
-
-    def __init__(self, slug: str):
-        for status in CONTENT_DEFAULT_STATUS:
-            if slug == status.slug:
-                super(ContentStatusLegacy, self).__init__(
-                    slug=status.slug,
-                    global_status=status.global_status,
-                    label=status.label,
-                    fa_icon=status.fa_icon,
-                    hexcolor=status.hexcolor,
-                )
-                return
+    def __init__(self, extend_content_status: typing.List[ContentStatus]):
+        self._content_status = [self.OPEN]
+        self._content_status.extend(extend_content_status)
+
+    def get_one_by_slug(self, slug: str) -> ContentStatus:
+        for item in self._content_status:
+            if item.slug == slug:
+                return item
         raise ContentStatusNotExist()
 
-    @classmethod
-    def all(cls, type='') -> ['NewContentStatus']:
-        return CONTENT_DEFAULT_STATUS
+    def allowed_slugs_values(self) -> typing.List[str]:
+        """ Get alls slugs"""
+        return [item.slug for item in self._content_status]
+
+    def allowed(self) -> typing.List[ContentStatus]:
```

Comment:
Why this method called "allowed" ? What is the motivation to not choose name like "get_all" ?

Reply:
That's right in this case allowed are same as get_all and that probably will not change as far i know , we can effectively rename it.

Reply:
Ok, do it.